### ISSUE TITLE
client.stop() does not change client.connecting state 

### ISSUE THREAD

**Issue by almakano** [isaackogan/TikTokLive#112]
I want to reconnect client, if any exception happen on "await client.start()"
doing client.stop() and sleep(20), but client.connecting keep remain True

**Comment by almakano** [comment]
Currently, to reconnect I found such solution:
```
client.stop()
client = TikTokLiveClient(client.unique_id)
sleep(20)
```

But, now, I have a challenge with InitialCursorMissing and SignatureRateLimitReached exceptions, which behavior I still invetigating

I run 10 tasks within 2 seconds timeout
And when online lives come in first 3 indexes on script restart, they connect succesfully. But if online lives placed between 5 and 10 indexes, they trap into InitialCursorMissing exception

**Comment by SKevo18** [comment]
> What about using OOP to allow people to overload certain attributes to give that additional control in a subclass?

It depends on your way to code and I don't want to interfere with it, but I'd personally go with the approach of adding an additional argument to `TikTokLiveClient.__init__`. That provides people with a setting that they can turn on/off at anytime, without the need to create a custom logic for it or subclassing that class, if I understood your solution correctly(?)

Of course, the final decision is up to you - you seem more skilled than me and you know your framework the best, so if your approach is better than mine, I'd say go with it!

**Comment by isaackogan** [comment]
Btw, do **not** use time.sleep(), it is a **blocking** operation. Here's something better:


```py
import asyncio
....

async def main(client: TikTokLiveClient):
        while ...:
            try:
                await client.start()
                break

            except LiveNotFound:
                warn(f"User `@{client.unique_id}` seems to be offline, retrying after 1 minute...")
                await asyncio.sleep(60)
```

**Comment by SKevo18** [comment]
Ah yes, that's correct... I shouldn't mix async and sync. As this still raises `TikTokLive.types.errors.AlreadyConnecting`, I would like to find a workaround for the time being until it's patched

Do you mind if we chat at Discord? I don't want to clog this issue with non-relevant stuff, and I may need an occasional help hand while developing my project. You can add me if you want: `Kevo#7590`

**Comment by isaackogan** [comment]
I infrequently use Discord, your best bet is to talk over GitHub... Hopefully almakano doesn't mind :D.

As for this version, to put it lightly, you're fricked. My logic is awful.

You will want to override `_websocket_connect` in a subclass and remove the part that sets self.__connecting to True. Note that this will mean that you lose a bit of safety on checking if you are connecting for multithreaded stuff. But that's probably OK in your use case.

Isaac

**Comment by SKevo18** [comment]
> I infrequently use Discord, your best bet is to talk over GitHub... Hopefully almakano doesn't mind :D.

That's okay, I understand :)

> You will want to override `_websocket_connect` in a subclass and remove the part that sets self.__connecting to True. Note that this will mean that you lose a bit of safety on checking if you are connecting for multithreaded stuff. But that's probably OK in your use case.

I see. It indeed seems that this part of the code screws things up:

https://github.com/isaackogan/TikTokLive/blob/863ac2fd69cf9ebe40b9539c99a9ae93c47cb14d/TikTokLive/client/base.py#L290-L294

Since subclassing and overwriting it is difficult because of Python's variable mangling for private methods, I took the simpler route of removing it directly in the source code of the library in my virtualenv. So far, it doesn't seem to have any side-effects(?) I will keep it running and let you know if it works. For reference, this is the current code that I use in my subclass (minus the code linked above):

```python
    async def main(self):
            while not self.connected:
                try:
                    await self.start()
    
                except LiveNotFound:
                    print(f"User `@{self.unique_id}` seems to be offline, retrying after 1 minute...")
    
                    await asyncio.sleep(60)
```

**Comment by chaseoes** [comment]
I am also having this problem.

**Comment by isaackogan** [comment]
Just want to mention:

> Since subclassing and overwriting it is difficult because of Python's variable mangling for private methods, I took the simpler route of removing it directly in the source code of the library in my virtualenv. So far, it doesn't seem to have any side-effects(?) I will keep it running and let you know if it works. For reference, this is the current code that I use in my subclass (minus the code linked above):

Since this method has one underscore, it is just protected, so subclassing and overriding works, easily.

**Comment by Geczy** [comment]
@SKevo18 found your repo but your code seems to have some issue still, even with those already connected lines removed from the venv. Did you ever revisit this?

**Comment by SKevo18** [comment]
> @SKevo18 found your repo but your code seems to have some issue still, even with those already connected lines removed from the venv. Did you ever revisit this?

No, sorry, my repository was never finished nor is being actively used, so it is possible that the code is not working anymore in an ever changing environment that TikTok's API is.

**Comment by Geczy** [comment]
@SKevo18 all good, i found an alternative that works even better, and has a UI. i'm using it to record all my lives i need at once 

https://github.com/go-olive/olive

**Comment by isaackogan** [comment]
Fixed in v6.0.0
